nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -86,75 +87,86 @@ def __init__(
             fine-tuning the quantization algorithm
         """
         super().__init__()
+        self.preset = preset
+        self.target_device = target_device
+        self.subset_size = subset_size
+        self.fast_bias_correction = fast_bias_correction
+        self.model_type = model_type
+        self.ignored_scope = ignored_scope
+        self.dump_intermediate_models = dump_intermediate_models
+        self.advanced_parameters = advanced_parameters
+        self.reset()
+
+    def reset(self):
```

Comment:
Why do you need to reset algorithms?

Reply:
Reverted. Added a logic into MinMax to reset inner state when a new model comes for quantization